benchmark/single-source/RomanNumbers.swift

```diff
@@ -0,0 +1,99 @@
+//===--- RomanNumbers.swift -----------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import TestsUtils
+
+//
+// Mini benchmark implementing roman numeral conversions to/from integers.
+// Measures performance of Substring.starts(with:) and String.append(),
+// with very short string arguments.
+//
+
+public let RomanNumbers = [
+  BenchmarkInfo(
+    name: "RomanNumbers",
+    runFunction: run_RomanNumbers,
+    tags: [.api, .String, .algorithm])
+]
+
+let romanTable: [(Int, String)] = [
+  (1000, "M"),
+  (900, "CM"),
+  (500, "D"),
+  (400, "CD"),
+  (100, "C"),
+  (90, "XC"),
+  (50, "L"),
+  (40, "XL"),
+  (10, "X"),
+  (9, "IX"),
+  (5, "V"),
+  (4, "IV"),
+  (1, "I"),
+]
+
+extension BinaryInteger {
+  var romanNumeral: String {
+    var result = ""
+    var value = self
+  outer:
+    while value > 0 {
+      var position = 0
+      for i in position ..< romanTable.count {
```

Comment:
I'd prefer something like `for (i, (v, s) in romanTable.enumerated() {`

(or even `in zip(romanTable.indices, romanTable)`)

Whether that makes for a better benchmark is another matter :)